app/models/metric/ci_mixin/long_term_averages.rb

```diff
@@ -10,22 +10,29 @@ module Metric::CiMixin::LongTermAverages
     end
   end
 
+  def generate_vim_performance_operating_ranges
+    TimeProfile.all.each { |tp| generate_vim_performance_operating_range(tp) }
+  end
+
   private
 
-  def averages_over_time_period(col, typ)
-    # there is only ever 1 of these. It has days = 30 (AVG_DAYS)
-    vpor = vim_performance_operating_ranges.detect do |rec|
-      rec.days == Metric::LongTermAverages::AVG_DAYS
-    end
+  def generate_vim_performance_operating_range(time_profile)
+    vpor = vim_performance_operating_ranges
+           .create_with(:days => Metric::LongTermAverages::AVG_DAYS)
+           .find_or_create_by(:time_profile => time_profile)
+    vpor.recalculate_values
+    vpor.save!
+  end
 
-    if vpor.nil? || vpor.updated_at.utc < 1.day.ago.utc
-      vpor ||= vim_performance_operating_ranges.build(:days => Metric::LongTermAverages::AVG_DAYS)
-      options = {:avg_days => Metric::LongTermAverages::AVG_DAYS}
-      averages = Metric::LongTermAverages.get_averages_over_time_period(self, options)
+  def averages_over_time_period(col, typ)
+    vpor = vim_performance_operating_ranges.first # All vpors have 30 days
+    # TODO do we deal with time profiles here?
```

Comment:
> So prior to 2d3e2f7, that never passed in a option hash to this method (when it used to take one) so the value for that .time_profile (not id... which means it had to make a SQL call each time we went through that if one existed...) would aways have to be nil. This was removed in the aforementioned commit, which changed the functionality and probably shouldn't have.

Ohhhh now I understand... Before 2d3e2f7 , a time_profile option existed, but it was erroneously never used, and thus was removed in 2d3e2f7 . That's good...then I can put it back and make sure we pick up the right one.  Then somehow we'll have to figure out a way to pass the TimeProfile down.